controller/api/public/stat_summary.go

```diff
@@ -227,47 +214,125 @@ func (s *grpcServer) objectQuery(
 		},
 	}
 
-	return &rsp, nil
+	return resourceResult{res: &rsp, err: nil}
 }
 
+func (s *grpcServer) nonK8sResourceQuery(ctx context.Context, req *pb.StatSummaryRequest) resourceResult {
+	requestMetrics, err := s.getPrometheusMetrics(ctx, req, req.TimeWindow)
+	if err != nil {
+		return resourceResult{res: nil, err: err}
+	}
+	rows := make([]*pb.StatTable_PodGroup_Row, 0)
+
+	for resourceKey, metrics := range requestMetrics {
+		resource := resourceKey
+		resource.Type = req.GetSelector().GetResource().GetType()
+
+		row := pb.StatTable_PodGroup_Row{
+			Resource:   &resource,
+			TimeWindow: req.TimeWindow,
+			Stats:      metrics,
+		}
+		rows = append(rows, &row)
+	}
+
+	rsp := pb.StatTable{
+		Table: &pb.StatTable_PodGroup_{
+			PodGroup: &pb.StatTable_PodGroup{
+				Rows: rows,
+			},
+		},
+	}
+	return resourceResult{res: &rsp, err: nil}
+}
+
+func isNonK8sResourceQuery(resourceType string) bool {
+	isNonK8s := false
+	if resourceType == k8s.Authority {
+		isNonK8s = true
+	}
+
+	return isNonK8s
+}
```

Comment:
TIOLI You could simplify this func to:

```
func isNonK8sResourceQuery(resourceType string) bool {
	return resourceType == k8s.Authority
}
```

That's less extensible if we add more non-k8s resource types though.